code/modules/paperwork/papershredder.dm

```diff
@@ -47,6 +47,8 @@
 				paperamount += paper_result
 			qdel(W)
 			playsound(src.loc, 'sound/items/pshred.ogg', 75, 1)
+			user << "<span class='notice'>You shred the paper.</span>"
+			add_overlay("papershredder_on")
```

Comment:
I think that using `flick()` would be better

```suggestion
			flick("papershredder_on", src)
```

Reply:
the problem using flick is that it overrides the paper shredder sprite instead of overlaying it

and will cause it to temporarily display an empty paper shredder instead of showing the papershredder's fullness alongside the animation

Reply:
If you want it to play animation, then you can make it as an overlay, and then right after you add an overlay add a timer that calls update_icon() proc after x time.

With that you need to add cut_overlays() to the update_icon() so it get's rid of animation.